### DISCUSSION TITLE
【教程】各种软件报错的通用处理办法

### DISCUSSION THREAD

**Discussion by capacitor1** [share121/inter-knot#1850] [其他]
![image](https://github.com/user-attachments/assets/49125c46-d347-43df-8609-61856f7929bc)

## 计算机软件报错的通用处理办法（仅Windows）

*注：不保证100%解决问题。但是大概率思路和方向是对的*

自己写的，有什么问题欢迎评论。

### 1 找不到xxx.dll，或xxx.dll not found

- 重新安装该软件

- 从网上下载对应的dll文件并放在正确的地方。

### 2 xxx静态程序接入点错误（xxx.dll@xx），并且跟着一堆乱码

- dll的版本和程序不符。重新安装软件，或者检查软件更新时是否不正常的中断过。

- 如果是问题1引发的，那需要重新查找另一个版本的dll来尝试

### 3 xxx Access Denied，并且xxx是一个文件

- 找到对应的文件，如果文件不存在，则手动创建一个

- 使用管理员模式重新运行该软件

- 检查文件所在的安装路径是否具有什么不该有的权限（尝试删除或打开对应文件，看看是否能操作），并删除掉权限

### 4 network error 或者 无法链接到远程服务器xxx:443 或者 connect timeout

- 重新连接一个可以上网的网络。

- 如果是国外程序，确保VPN配置到位并可用。

- 不要下载过于古老的版本，因为旧版本软件的服务器可能已经不存在。尝试更新软件

- 不要连接到网速过于慢的网络。选择一个速度达标的网，再运行软件。

### 5 xxx（一个文件）not found

- 手动创建对应的文件。

- 重新安装软件

- **如果文件路径能看出明显的乱码，那么需要重命名为全英文路径，再运行软件**

### 6 文件写入失败：指定不存在的设备 或者 遇到一个不可预料的错误，但是对应的设备是存在的，并且对应的文件也存在

> **这种情况可能比较严重，大概率是因为硬盘损坏导致的无法写入**

- 立即使用HDT等软件检查磁盘是否有损坏的扇区。如果有，则立即更换硬盘

- （如果是外接硬盘）检查连接线是否接触不良，或者装置温度过高

### 7 SSL Protocol Error等与SSL有关的错误

- 切换网站访问方式，不要使用https，尝试使用http直接访问

- 刷新网站缓存

- 检查本地时间是否正常，重新校对时间

### 8 VPN软件，连接显示403而不是timeout 或者 延迟很低却连不上

- 检查VPN流量是否耗尽，并更新节点

- 检查本地时间是否正常，时间与GMT时间误差不得大于10s。重新校对时间

- 检查订阅连接是否更新了，刷新一下那个订阅

### 9 复制/剪切/修改文件显示 需要来自xxx的权限，并且无论怎样都没办法复制。

- 不要动Windows的系统文件。系统文件可能受保护

- 有可能被杀毒软件锁定了。在杀毒软件界面里，把对应文件添加白名单，或者关闭杀毒软件

- 检查文件是不是被强制删除过，但不成功。（这种情况下，文件属性为T）使用diskgenius等软件直接清扇区

### 10 访问网站，返回5xx错误，并且访问不了

- 5xx是服务器错误，不是你的问题。停止访问此网站

### 11 访问网站被CF拦截显示you have been blocked

- 对于部分网站，不能使用日本节点访问，建议用新加坡或者美国节点

- 如果是API访问，检查是不是发请求的频率太快了。等几个小时再访问，或者换一个IP再访问

- 也有可能是服务器配置的问题。这种情况需要等站长修复

### 12 解压压缩包，观看视频时报错：unsupported method 或者unknown encoding错误，并且无法继续。排除了类似于问题3的情况

> 使用的软件不支持这种格式。更换一个更完善的软件

- 对于音视频播放，系统自带的播放器无法播放几十年前的老片。建议使用potplyer/vlc等播放器

- 对于压缩包，建议使用7-zip-zstd解压。或者使用这种格式专用的软件解压

### 13 解压压缩包，文件名全乱码

- 日本计算机处理的压缩包传到中国会有这情况。需要更改解压软件的编码到S-JIS

- 检查压缩包是否损坏，是的话重新下载压缩包

### 14 运行一些GitHub上的软件，闪了一个黑屏界面就退出了

- 这些软件可能是命令行软件，需要用cmd来执行

- 检查是不是和问题1一样少下载了dll

- 检查是不是问题5

### 15 运行一些命令行软件，页面全是乱码的

- bat脚本可以用chcp来切换编码

- 不要用太古老的软件。确保软件是最新版并且文件没有损坏

- 不建议运行未经汉化的日本软件。去浏览器搜索一个汉化版来用

> 如果以上方法都不行，可以这样：把乱码复制一下，粘贴到hex编辑器里（一定要用hex编辑器，不能用word文档或者记事本），然后再用支持调整文本编码的软件打开刚刚粘贴的内容。

### 16 在虚拟机中运行系统，硬盘没满却报错空间不足

- 大概率是物理机硬盘满了

### 17 使用ffmpeg转码视频，一回车就报错

- 首先看看命令是否输入错误。不能少空格或多空格。输入文件要存在。参数要正确

- 检查输入文件的编码是否支持。但实际上ffmpeg大概率都支持

- 检查输入文件是不是损坏了。比如输入的文件根本就不是视频，但却是mp4后缀

- 检查目标编码是否支持。ffmpeg只是支持输入各种格式，但是支持输出的格式非常有限。 比如vp6 vp8就不支持输出

### 18 使用ffmpeg转码视频，能够处理，但是中间不停的报错：bitstream error，或者timestamp error

- 视频中间可能有损坏。如果视频播放时候花屏，就是损坏了。尝试重新下载视频。

- 如果是转码直播HLS录制的视频，有关timestamp的问题是正常的。如果报这个错，输入的视频大概率拖动不了进度条

- 如果是BT下载的视频，尝试重新校验一下分块

### 19 某些软件，不显示报错信息。但是一直是卡死无响应的状态 也不报错也不闪退

- 如果是文件处理用的，是不是一次给太多文件了。这种情况建议观察任务管理器：

  - 任务管理器观察这个无响应的软件是否正在读写硬盘或者占用cpu。并且占用的是100%，导致软件无响应

  - 如果是，那就是正常的

- 如果任务管理器查看发现，占用全是0%，还无响应，那就是软件自己出错了。建议结束进程

**Comment by Dave-12138** [comment]
>找不到xxx.dll，或xxx.dll not found

可别真在网上下一个放过去啊，正常做法是找这个动态链接库是哪个依赖库提供的，装上依赖库。
